Server/Plugins/APIDump/APIDesc.lua

```diff
@@ -11689,6 +11797,40 @@ end
 					},
 					Notes = "Returns true if the specified player is queued to be transferred to a World.",
 				},
+				RegisterForgeMod =
+				{
+					Params =
+					{
+						{
+							Name = "ModName",
+							Type = "string",
+						},
+						{
+							Name = "ModVersion",
+							Type = "string",
+						},
+					},
+					Notes = "Add a Forge mod name/version to the server ping list.",
+				},
+				RegisterForgeModForProtocol =
+				{
+					Params =
+					{
+						{
+							Name = "ModName",
+							Type = "string",
+						},
+						{
+							Name = "ModVersion",
+							Type = "string",
+						},
+						{
+							Name = "ProtocolVersionNumber",
+							Type = "number",
+						},
+					},
+					Notes = "Add a Forge mod name/version to the server ping list for all protocol versions.",
```

Comment:
I think so, my reasoning is the Forge mod version is inherently game-version specific (some mods have taken to putting the game version into their version itself, for example ironchests is version "1.12-7.0.31.818"). It remains to be seen but the way I envision it, there would be one Cuberite plugin, supporting multiple client mod versions, since there may be only minor or even no differences between them in a given mod. And the mod version has to be correct (different for each game version) for the client to show the server as compatible.

Maybe there's another way to accomplish this. Instead of a per-protocol mod concept (where the mod won't even show up on other protocol versions), maybe the register mod function could take the mod name, and then a map of protocol version to mod version? On the other hand, per-protocol mods may allow for smoother upgrade transitions: if a mod isn't supported for or applicable to a version, it won't even appear in that version's ModList.